bot/cogs/utils/api.py

```diff
@@ -20,133 +38,154 @@ def __init__(self, id, ip, port, web_url=None):
-        return f'connect {self.ip}:{self.port}'
+        """
+        Format console command to connect to server.
+        """
+
+        return "connect {}:{}".format(self.ip, self.port)
 
     @property
     def match_page(self):
-        """ Generate the matches CS:GO League page link. """
-        if self.web_url:
-            return f'{self.web_url}/match/{self.id}'
-
-
-async def start_request_log(session, ctx, params):
-    """"""
-    ctx.start = asyncio.get_event_loop().time()
-    logger = logging.getLogger('csgoleague.api')
-    logger.info(f'Sending {params.method} request to {params.url}')
-
-
-async def end_request_log(session, ctx, params):
-    """"""
-    logger = logging.getLogger('csgoleague.api')
-    elapsed = asyncio.get_event_loop().time() - ctx.start
-    logger.info(f'Response received from {params.url} ({elapsed:.2f}s)\n'
-                f'    Status: {params.response.status}\n'
-                f'    Reason: {params.response.reason}')
-    resp_json = await params.response.json()
-    logger.debug(f'Response JSON from {params.url}: {resp_json}')
-
-
-class ApiWrapper:
-    """ Class to contain API request wrapper functions. """
-
-    def __init__(self, loop, base_url, api_key):
-        """ Set attributes and initialize logging handlers. """
-        # Set attributes
-        self.base_url = base_url
-        self.api_key = api_key
-        self.logger = logging.getLogger('csgoleague.api')
-
-        # Check API URL
-        if not self.base_url.startswith('https') and self.base_url.startswith('http'):
-            self.logger.warning(f'API url "{self.base_url}" should start with "https" instead of "http"')
-
-        # Register trace config handlers
-        trace_config = aiohttp.TraceConfig()
-        trace_config.on_request_start.append(start_request_log)
-        trace_config.on_request_end.append(end_request_log)
-
-        # Start session
-        self.logger.info('Starting API helper client session')
-        self.session = aiohttp.ClientSession(loop=loop, json_serialize=lambda x: json.dumps(x, ensure_ascii=False),
-                                             raise_for_status=True, trace_configs=[trace_config])
-
-    async def close(self):
-        """ Close the API helper's session. """
-        self.logger.info('Closing API helper client session')
-        await self.session.close()
+        """
+        Generate the matches CS:GO League page link.
+        """
 
-    @property
-    def headers(self):
-        """ Default authentication header the API needs. """
-        return {'authentication': self.api_key}
+        if Config.api_url:
+            return f'{Config.api_url}/match/{self.id}'
 
-    async def generate_link_url(self, user_id):
-        """ Get custom URL from API for user to link accounts. """
-        url = f'{self.base_url}/discord/generate/{user_id}'
 
-        async with self.session.get(url=url, headers=self.headers) as resp:
+class User:
```

Comment:
I'm worried that having another User class would be confusing. Discord.py already has one that appears everywhere. It would be nice if we could subclass Discord.py's User class to add these methods, but we couldn't enforce the object type they give to us without monkey patching it into their library.

Not sure if there are any good alternatives for that, but at the very least I think it would make more sense to rename this to Player class and the current Player class can become a Stats class. Let me know what you think.

Reply:
Yeah agreed

Reply:
Yeah changing the current `Player`` class to `PlayerStats` & changing `User` to `Player` would be less confusing.